### PR TITLE
Adding CRUD tests for Compute Profile

### PR DESCRIPTION
**Pull request by oshtaier**
```
nosetests tests/foreman/api/test_computeprofile.py
.....
----------------------------------------------------------------------
Ran 5 tests in 90.875s

OK
```

### CODE REVIEW

**Comment by oshtaier** [comment]
Closes #3267

File: tests/foreman/api/test_computeprofile.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        for name in invalid_values_list():
+            with self.subTest(name):
+                with self.assertRaises(HTTPError):
+                    entities.ComputeProfile(name=name).create()
+
+    @tier1
+    def test_positive_update_name(self):
+        """Update selected Compute Profile entity using proper names
+
+        @Feature: Compute Profile - Positive Update
```
Comment (sthirugn) [inline tests/foreman/api/test_computeprofile.py]: Naming the `@Feature` field as `Positive Create`, `Negative Create`, `Positive Update`, `Negative Update` is kind of duplicating the same info we put in test case names.  We may just call it `Compute Profile`.  Anyways this is just a food for thought.  Not a blocker for this PR.

Comment (oshtaier) [inline tests/foreman/api/test_computeprofile.py]: I like your idea and actually wanted to have implementation like that in the past, but decided to follow overall style. Don't mind to discuss the idea in future

**Comment by sthirugn** [comment]
ACK

**Comment by jyejare** [comment]
Are you missing 'run_only_on('sat')' decorator ?

File: tests/foreman/api/test_computeprofile.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+        @Feature: Compute Profile - Negative Create
+
+        @Assert: Compute Profile is not created
+        """
+        for name in invalid_values_list():
+            with self.subTest(name):
+                with self.assertRaises(HTTPError):
+                    entities.ComputeProfile(name=name).create()
+
```
Comment (jyejare) [inline tests/foreman/api/test_computeprofile.py]: Here, You may actually validate the creation of compute profile by searching with its name.

**Comment by jyejare** [comment]
ACK pending comment, question.

File: tests/foreman/api/test_computeprofile.py (hunk 2 above)
Comment (oshtaier) [inline tests/foreman/api/test_computeprofile.py]: Not sure what you mean. That is negative scenario which actually produces an exception and doesn't populate that object at all. Here, we performing usual validation which is used across all API test cases

**Comment by oshtaier** [comment]
Addressed all comments

File: tests/foreman/api/test_computeprofile.py (hunk 2 above)
Comment (jyejare) [inline tests/foreman/api/test_computeprofile.py]: @oshtaier : After producing exception we still want to see if its not actually creating Compute Profile. Only Exception wont be enough to validate its not created. We cant go with the assumption as you said 'and doesn't populate ComputeProfile object at all'.

**Comment by rplevka** [comment]
Very decent PR :+1: 
ACK pending @jyejare's comment

File: tests/foreman/api/test_computeprofile.py (hunk 2 above)
Comment (rplevka) [inline tests/foreman/api/test_computeprofile.py]: @jyejare +1

Comment (sthirugn) [inline tests/foreman/api/test_computeprofile.py]: @jyejare For many/most negative test cases, we just check the error now. Your suggestion is great and it is something which needs to be implemented across robottelo and not just this PR.

**Comment by rplevka** [comment]
@jyejare, i'm ok with @sthirugn's reaction and am more than happy to get this merged as is now. ACK+Merge if you're ok with this.

**Comment by jyejare** [comment]
Ok. We can create a new issue for updating the negative test cases overall as per my comment https://github.com/SatelliteQE/robottelo/pull/3271#discussion_r50966272.

As of now I am merging this PR.

ACK, Thanks.

**Comment by oshtaier** [comment]
Mentioned issue should be addressed in https://github.com/SatelliteQE/robottelo/issues/3273

**Comment by sthirugn** [comment]
Thank you @jyejare and @rplevka for acking my comments.

@oshtaier :+1: for creating a new issue.

**Pull request merged**